Integrations/integration-UrlScan.yml

```diff
@@ -0,0 +1,411 @@
+            }
+            for (var i = 0; i < resStats.resourceStats[0].countries.length-1; i++){
+                md += " **" + resStats.resourceStats[0].countries[i] + ",**";
+            }
+            if (resStats.resourceStats[0].countries.length > 1) {
+                md += " **" + resStats.resourceStats[0].countries[resStats.resourceStats[0].countries.length-1] + ")**\n";
+            }
+            if (resStats.ipStats[0].asn !== undefined) {
+                md += "ASN: **" + resStats.ipStats[0].asn.asn + " " + resStats.ipStats[0].asn.description + "**\n";
+            }
+            if (resStats.malicious !== undefined) {
+                md += "Malicious: **" + resStats.malicious + "**";
+            }
+
+            if (resStats.malicious !== 0){
+                dbotScore = 3;
+                ec.URL = {};
+                addMalicious(ec, "URL", {
+                    Data: url,
+                    Malicious: {Vendor: 'UrlscanIO', Description: 'Match found in Urlscan.io database'}
+                });
+            }else{
+                dbotScore = 1;
+            }
+
+            ec.DBotScore = {Indicator: args.url, Type: 'url', Vendor: 'Urlscan.io', Score: dbotScore};
+            resDis.push ( {'ContentsFormat': formats.json, 'Type': entryTypes.note, 'Contents': res, 'HumanReadable': md, 'EntryContext': ec} );
+            resDis.push(getScreenshot(res.results[0]._id));
+
+            return resDis;
+
+        }else{
+            ec.DBotScore = {Indicator: args.url, Type: 'url', Vendor: 'UrlscanIO', Score: dbotScore};
+            md += "No results for this url";
+            resDis.push( {'ContentsFormat': formats.json, 'Type': entryTypes.note, 'Contents': res, 'HumanReadable': md, 'EntryContext': ec} );
+        }
+
+        return resDis;
+    };
+
+    var getScreenshot = function(id) {
+        var scanRes = searchResults(id);
+        if (scanRes.status === 404) {
+            return scanRes.description;
+        }
+        var img = scanRes.task.screenshotURL;
+        var res = sendRequest('GET', '', '','' , img);
+        var fileId = saveFile(res);
+
+        return {'Contents': '', 'ContentsFormat': formats.text, 'Type': 7, 'File': id, 'FileID': fileId };
+    };
+
+    var searchIP = function(ip) {
+        var res = sendRequest("GET","/search?q=ip:" + ip);
+        var md = "";
+        var ec = {};
+        var resDis = [];
+        var dbotScore = 0;
+
+        md += "## Urlscan.io Reputation for " + ip + "\n";
+
+        if (res.results.length !== 0){
+            var resStats = searchResults(res.results[0]._id).stats;
+
+            if (resStats.securePercentage !== undefined) {
+                md += "Security Percentage: **"+resStats.securePercentage + "**\n";
+            }
+            if (resStats.resourceStats[0].countries !== undefined) {
+                md += "Number of countries associated with this site: **" + resStats.resourceStats[0].countries.length + " (**";
+            }
```

Comment:
this look a bit repetitive. Not sure it is the same, but you can look at alien vault for a less repetitive code doing something similar